### PR TITLE
Extend Connect() API with an optional compress flag.

### PR DESCRIPTION
**Pull request by gxben**
This allows for clients to decide whether or not they want
HTTP connections to vSphere server to be made with
support for gzip,deflate support (hence being compressed).

Once enabled, it appears that some requests are sent by Web
server as HTTP chunks (Transfer-Encoding = chunked),
which appears not be supported by PyVmomi SDK so far.

Disabling HTTP compression support provides better
compatibility against all kind of server responses.

### CODE REVIEW

**Comment by hartsock** [comment]
I like this. Any shot of developing a pair of tests to show that the flag works and that it does no harm when not set?

**Comment by hartsock** [comment]
* I like this
* I need tests
* We need to consider other issues like #212

File: pyVim/connect.py [hunk 1]
```diff
@@ -532,7 +537,7 @@ def __FindSupportedVersion(protocol, server, port, path, preferredApiVersions):
 
 def SmartConnect(protocol='https', host='localhost', port=443, user='root', pwd='',
                  service="hostd", path="/sdk",
-                 preferredApiVersions=None):
+                 preferredApiVersions=None, compress=True):
```
Comment (hartsock) [inline pyVim/connect.py]: Default behaviors should always be "whatever you did before" ... SoapStubAdapter currently defaults to compressed responses ... so this is actually correct.

**Comment by hartsock** [comment]
TODO: fork this PR, rebase, then merge a clean commit history before Merge.

**Comment by tianhao64** [comment]
PyVmomi supports gzip and deflated connections. Plus I don't see how this change disabled these flags. Closing.

**Pull request closed without merging**